src/v/raft/replicate_batcher.h

```diff
@@ -43,6 +43,15 @@ class replicate_batcher {
             if (_replicate_opts.timeout) {
                 _timeout_timer.arm(_replicate_opts.timeout.value());
             }
+            if (_replicate_opts.as) {
+                auto& as = _replicate_opts.as->get();
+                if (as.abort_requested()) {
+                    mark_as_aborted();
+                } else {
+                    _abort_sub = as.subscribe(
```

Comment:
nit: I think the idiomatic way of doing this is like..

```
_abort_sub = as.subscribe( { [] () { foo;}});
if (!_abort_sub) {
  foo();
} 
```

Reply:
i will refactor after, would be great to merge this to see if fixed

Reply:
for sure, not a blocker.